### PR TITLE
Fix test_connect_ipv6_addr.

### PR DESCRIPTION
**Pull request by RatanShreshtha**
I have added a check in for ipv6 host in `_build_tunnel_request` and accordingly modify the target, is there a better way ?

### CODE REVIEW

**Comment by njsmith** [comment]
Hmm. I bet we're also going to need a similar check somewhere to set up the `Host:` header correctly when doing a regular (no-proxy) request to a raw IPv6 address...

File: src/urllib3/_async/connection.py [hunk 1]
```diff
@@ -210,7 +210,13 @@ def _build_tunnel_request(host, port, headers):
     Builds a urllib3 Request object that is set up correctly to request a proxy
     to establish a TCP tunnel to the remote host.
     """
-    target = "%s:%d" % (host, port)
+
+    try:
+        socket.inet_pton(socket.AF_INET6, host)
+        target = "[%s]:%d" % (host, port)
+    except OSError:
+        target = "%s:%d" % (host, port)
```
[the remark this answers is not in the document]
Comment (RatanShreshtha) [reply src/urllib3/_async/connection.py]: thanks it seem more readable to me.

**Comment by RatanShreshtha** [comment]
I will look where we need to set-up such check for `Host: ` header.

**Comment by pquentin** [comment]
> Hmm. I bet we're also going to need a similar check somewhere to set up the Host: header correctly when doing a regular (no-proxy) request to a raw IPv6 address...

That does not exist at all in the urllib3 tests currently, so maybe we can address this in a future pull request? I opened https://github.com/python-trio/urllib3/issues/87 to this effect

**Pull request merged**